services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProvidersResource.java

```diff
@@ -0,0 +1,139 @@
+import org.keycloak.services.resources.admin.IdentityProvidersResource;
+import org.keycloak.services.resources.admin.permissions.AdminPermissionEvaluator;
+
+@Provider
+public class OrganizationIdentityProvidersResource {
+
+    private final KeycloakSession session;
+    private final RealmModel realm;
+    private final OrganizationProvider organizationProvider;
+    private final OrganizationModel organization;
+    private final AdminPermissionEvaluator auth;
+    private final AdminEventBuilder adminEvent;
+
+    public OrganizationIdentityProvidersResource() {
+        // needed for registering to the JAX-RS stack
+        this(null, null, null, null);
+    }
+
+    public OrganizationIdentityProvidersResource(KeycloakSession session, OrganizationModel organization, AdminPermissionEvaluator auth, AdminEventBuilder adminEvent) {
+        this.session = session;
+        this.realm = session == null ? null : session.getContext().getRealm();
+        this.organizationProvider = session == null ? null : session.getProvider(OrganizationProvider.class);
+        this.organization = organization;
+        this.auth = auth;
+        this.adminEvent = adminEvent;
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response addIdentityProvider(IdentityProviderRepresentation providerRep) {
+        auth.realm().requireManageRealm();
+
+        Response response = new IdentityProvidersResource(realm, session, auth, adminEvent).create(providerRep);
+
+        if (Status.CREATED.getStatusCode() == response.getStatus()) {
+            try {
+                IdentityProviderModel identityProvider = realm.getIdentityProviderByAlias(providerRep.getAlias());
+
+                if (organizationProvider.addIdentityProvider(organization, identityProvider)) {
+                    return response;
+                }
+
+                throw ErrorResponse.error("Identity provider already associated to the organization", Status.BAD_REQUEST);
+            } catch (ModelException me) {
+                throw ErrorResponse.error(me.getMessage(), Status.BAD_REQUEST);
+            }
+        }
+
+        return response;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<IdentityProviderRepresentation> getIdentityProviders() {
+        auth.realm().requireManageRealm();
+        return organization.getIdentityProviders()
+                .map(this::toRepresentation)
+                .toList();
+    }
+
+    @Path("{alias}")
+    public IdentityProviderResource getIdentityProvider(@PathParam("alias") String alias) {
+        IdentityProviderModel broker = realm.getIdentityProviderByAlias(alias);
+        return new IdentityProviderResource(auth, realm, session, broker, adminEvent) {
+            @Override
+            public Response delete() {
+                Response response = super.delete();
+
+                if (organizationProvider.removeIdentityProvider(organization, broker)) {
```

Comment:
`organizationProvider.removeIdentityProvider(organization, broker)` should remove the IdP from realm ... shouldn't it be already removed by `super.delete();`?

Reply:
Yeah, and we are removing both delete and update operations from this endpoint. Similarly, as we did for users. That also means only unlinking the broker from the org instead of removing it entirely.

For now, I would keep as is.

Reply:
I'm afraid I do not understand, my point was that the IdP is being removed twice, is that intentional?